flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecMultipleInput.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.nodes.exec.stream;
+
+import org.apache.flink.api.dag.Transformation;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.planner.delegation.StreamPlanner;
+import org.apache.flink.table.planner.plan.nodes.exec.ExecNode;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Stream exec node for multiple input which contains a sub-graph of {@link ExecNode}s.
+ * The root node of the sub-graph is {@link #outputNode}, and the leaf nodes of the sub-graph are
+ * the output nodes of the {@link #getInputNodes()}.
+ *
+ * <p>The following example shows a graph of {@code ExecNode}s with multiple input node:
+ * <pre>{@code
+ *          Sink
+ *           |
+ * +---------+--------+
+ * |         |        |
+ * |       Join       |
+ * |     /     \      | BatchExecMultipleInput
+ * |   Agg1    Agg2   |
+ * |    |       |     |
+ * +----+-------+-----+
+ *      |       |
+ * Exchange1 Exchange2
+ *      |       |
+ *    Scan1   Scan2
+ * }</pre>
+ *
+ * <p>The multiple input node contains three nodes: `Join`, `Agg1` and `Agg2`.
+ * `Join` is the root node ({@link #outputNode}) of the sub-graph,
+ * `Agg1` and `Agg2` are the leaf nodes of the sub-graph,
+ * `Exchange1` and `Exchange2` are the input nodes.
+ */
+public class StreamExecMultipleInput extends StreamExecNode<RowData> {
+
+	private final ExecNode<?> outputNode;
+
+	public StreamExecMultipleInput(
+			List<ExecNode<?>> inputNodes,
+			ExecNode<?> outputNode,
```

Comment:
RowType output instead of outputNode, would be enough I think.

Reply:
`outputNode` is used for translating the sub-graph into transformation